spec/models/reports/facility_state_spec.rb

```diff
@@ -67,4 +59,103 @@
+        create(:patient, assigned_facility: facility, recorded_at: june_2021[:under_12_months_ago])
+        create(:patient, assigned_facility: facility, recorded_at: june_2021[:under_3_months_ago])
+        create(:patient, assigned_facility: facility, recorded_at: june_2021[:end_of_month])
+
+        RefreshMaterializedViews.new.refresh_v2
+        with_reporting_time_zone do
+          expect(described_class.find_by(facility: facility, month_date: june_2021[:now] - 2.years).cumulative_assigned_patients).to eq 1
+          expect(described_class.find_by(facility: facility, month_date: june_2021[:under_12_months_ago]).cumulative_assigned_patients).to eq 2
+          expect(described_class.find_by(facility: facility, month_date: june_2021[:under_3_months_ago]).cumulative_assigned_patients).to eq 3
+          expect(described_class.find_by(facility: facility, month_date: june_2021[:now]).cumulative_assigned_patients).to eq 4
+        end
+      end
+    end
+  end
+
+  context "treatment outcomes in the last 3 months" do
+    it "computes totals for under care patients" do
+      facility = create(:facility)
+
+      patient_controlled = create(:patient, assigned_facility: facility, recorded_at: june_2021[:long_ago])
+      create(:bp_with_encounter, :under_control, patient: patient_controlled, recorded_at: june_2021[:now] - 1.month)
+
+      patient_uncontrolled = create(:patient, assigned_facility: facility, recorded_at: june_2021[:long_ago])
+      create(:bp_with_encounter, :hypertensive, patient: patient_uncontrolled, recorded_at: june_2021[:now] - 1.months)
+
+      patient_missed_visit = create(:patient, assigned_facility: facility, recorded_at: june_2021[:long_ago])
+      create(:bp_with_encounter, patient: patient_missed_visit, recorded_at: june_2021[:over_3_months_ago])
+
+      patient_visited_no_bp = create(:patient, assigned_facility: facility, recorded_at: june_2021[:long_ago])
+      create(:prescription_drug,
+        device_created_at: june_2021[:now] - 1.month,
+        facility: facility,
+        patient: patient_visited_no_bp,
+        user: patient_visited_no_bp.registration_user)
+      create(:blood_pressure, patient: patient_visited_no_bp, recorded_at: june_2021[:over_3_months_ago])
+
+      RefreshMaterializedViews.new.refresh_v2
+      with_reporting_time_zone do
+        facility_state_june_2021 = described_class.find_by(facility: facility, month_date: june_2021[:now])
+
+        expect(facility_state_june_2021.controlled_under_care).to eq 1
+        expect(facility_state_june_2021.uncontrolled_under_care).to eq 1
+        expect(facility_state_june_2021.missed_visit_under_care).to eq 1
+        expect(facility_state_june_2021.visited_no_bp_under_care).to eq 1
+        expect(facility_state_june_2021.patients_under_care).to eq 4
+        expect(facility_state_june_2021.patients_lost_to_follow_up).to eq 0
+      end
+    end
+
+    it "computes totals for patients lost to follow up" do
```

Comment:
I noticed that the assigned and LTFU specs are sort of timeline specs, whereas control and care state are not. For good coverage, can we add a single timeline spec that, perhaps, tracks one patient over time in a facility going from ineligible for control (recent regn), to controlled, to missed visit, to LTFU?

Reply:
Only the `cumulative_assigned_patients` would be a bit of a timeline spec since that's necessary to assert the "cumulative" aspect of it. The definition of care states is in patient_states, and we've already unit tested the care state changing through time there ya?

Reply:
Oh I see now. Are we confident enough that we'll be grabbing the correct month's data from patient states and putting them in the correct month in this table?